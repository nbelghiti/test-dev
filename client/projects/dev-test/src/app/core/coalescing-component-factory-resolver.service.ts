/* tslint:disable */
/**
 * See https://github.com/jonrimmer/angular-coalescing-component-factory-resolver/blob/master/src/app/coalescing-component-factory-resolver.service.ts
 * Discussion & bug ticket: https://github.com/angular/angular/issues/14324#issuecomment-481898762
 *
 * Entry components are only available within a lazy loaded module, breaking dynamic generation (e.g. dialogs). This monkey patch enables resolving entry components from lazy loaded modules as well.
 *
 *
 * Provide in root module:
 *
 * @NgModule({
 *   providers: [CoalescingComponentFactoryResolver]
 * })
 * class AppModule {
 *   constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
 *     coalescingResolver.init();
 *   }
 * }
 *
 *
 * Use in lazy loaded module:
 *
 *
 * @NgModule({})
 * export class LazyModule {
 *   constructor(
 *     coalescingResolver: CoalescingComponentFactoryResolver,
 *     localResolver: ComponentFactoryResolver
 *   ) {
 *     coalescingResolver.registerResolver(localResolver);
 *   }
 * }
 *
 *
 * How it works: When initialised, it injects the root app's ComponentFactoryResolver and monkey patches the resolveComponentFactory method
 * to call its own implementation. This implementation first tries resolving the component factory on all the registered lazy-module resolvers,
 * then falls back to the root app resolver (there's a bit of extra logic to avoid cyclic calls, but that's the gist).
 */
/* tslint:enable */

import {
	ComponentFactory,
	ComponentFactoryResolver,
	Injectable,
	Type
} from '@angular/core';

@Injectable()
export class CoalescingComponentFactoryResolver extends ComponentFactoryResolver {
	private rootResolve: (component: Type<any>) => ComponentFactory<any>;

	private inCall = false;

	private readonly resolvers = new Map<
		ComponentFactoryResolver,
		(component: Type<any>) => ComponentFactory<any>
	>();

	constructor(private readonly rootResolver: ComponentFactoryResolver) {
		super();
	}

	init() {
		this.rootResolve = this.rootResolver.resolveComponentFactory;
		this.rootResolver.resolveComponentFactory = this.resolveComponentFactory;
	}

	public registerResolver(resolver: ComponentFactoryResolver) {
		const original = resolver.resolveComponentFactory;
		this.resolvers.set(resolver, original);
	}

	public resolveComponentFactory = <T>(
		component: Type<T>
	): ComponentFactory<T> => {
		// Prevents cyclic calls.
		if (this.inCall) {
			return null;
		}

		this.inCall = true;
		try {
			const result = this.resolveInternal(component);
			return result;
		} finally {
			this.inCall = false;
		}
	}

	private resolveInternal = <T>(component: Type<T>): ComponentFactory<T> => {
		for (const [resolver, fn] of Array.from(this.resolvers.entries())) {
			try {
				const factory = fn.call(resolver, component);
				if (factory) {
					return factory;
				}
			} catch { }
		}

		return this.rootResolve.call(this.rootResolver, component);
	}
}
