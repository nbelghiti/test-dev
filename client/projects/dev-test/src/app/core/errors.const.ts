export const ToastrConfig = {
	maxOpened: 2,
	timeOut: 8000
};
