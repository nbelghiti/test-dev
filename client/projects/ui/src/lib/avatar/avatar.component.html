<span
	class="c-avatar"
	[ngClass]="{
		'is-compact': compact
	}"
>
	<abbr [title]="name">{{initials}}</abbr>
</span>
