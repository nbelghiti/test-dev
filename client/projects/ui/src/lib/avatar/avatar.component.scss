// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$avatar-size-base: 4.4rem;
$avatar-size-compact: 3.2rem;

// =============================================================================
// :: Avatar
// =============================================================================
.c-avatar {
	display: flex;
	justify-content: center;
	align-items: center;
	width: $avatar-size-base;
	min-width: $avatar-size-base;
	height: $avatar-size-base;

	font-size: $font-size-1x-large;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;
	text-transform: uppercase;
	text-align: center;

	border-radius: 50%;
	background-color: $background-color-grey-96;

	cursor: default;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-avatar {

	&.is-compact {
		width: $avatar-size-compact;
		min-width: $avatar-size-compact;
		height: $avatar-size-compact;

		font-size: $font-size-base;
	}
}
