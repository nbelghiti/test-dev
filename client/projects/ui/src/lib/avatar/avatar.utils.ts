export const getInitials = (name: string): string => {
	const names = name.split(' ');
	let initials = names[0].slice(0, 1);

	if (names.length > 1) {
		initials += names[names.length - 1].slice(0, 1);
	}

	return initials.toUpperCase();
};
