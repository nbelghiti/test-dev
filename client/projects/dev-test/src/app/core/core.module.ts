import { ErrorHandler, NgModule } from '@angular/core';

import { CoalescingComponentFactoryResolver } from './coalescing-component-factory-resolver.service';
import { ContextModule } from './context/context.module';
import { GlobalErrorHandler } from './global-error-handler.service';
import { RouterModule } from './router/router.module';
import { UserModule } from './user/user.module';

@NgModule({
	imports: [
		ContextModule,
		RouterModule,
		UserModule,
	],
	providers: [
		CoalescingComponentFactoryResolver,
		{
			provide: ErrorHandler,
			useClass: GlobalErrorHandler,
		},
	],
})
export class CoreModule {
	constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
		coalescingResolver.init();
	}
}
