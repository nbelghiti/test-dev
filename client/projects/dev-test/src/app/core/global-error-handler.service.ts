import { ErrorHandler, Inject, Injectable, Injector } from '@angular/core';
import { TranslateService } from '@ngx-translate/core';
import { ToastrService } from 'ngx-toastr';

@Injectable()
export class GlobalErrorHandler extends ErrorHandler {
	constructor(
		@Inject(Injector) private readonly injector: Injector,
		private translate: TranslateService
	) {
		super();
	}

	handleError(error) {
		this.translate.get('ERRORS__GENERAL').subscribe((message: string) => {
			this.toastr.error(message, null, { onActivateTick: true });
		});

		console.error(error);

		super.handleError(error);
	}

	/**
	 * Need to get ToastrService from injector rather than constructor injection to avoid cyclic dependency error
	 * https://github.com/scttcper/ngx-toastr/issues/327
	 */
	private get toastr(): ToastrService {
		return this.injector.get(ToastrService);
	}
}
