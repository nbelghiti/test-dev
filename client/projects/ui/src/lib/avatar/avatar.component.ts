import { Component, Input } from '@angular/core';

import { getInitials } from './avatar.utils';

@Component({
	selector: 'ui-avatar',
	templateUrl: './avatar.component.html',
	styleUrls: ['./avatar.component.scss']
})

export class AvatarComponent {
	@Input() public compact: boolean = false;

	public initials = '';

	@Input() set name(name: string) {
		if (name) {
			this.initials = getInitials(name);
		}
	}
}
