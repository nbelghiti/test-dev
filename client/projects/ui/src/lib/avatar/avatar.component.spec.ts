import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { AvatarComponent } from './avatar.component';
import { getInitials } from './avatar.utils';

describe('AvatarComponent', () => {
	let component: AvatarComponent;
	let fixture: ComponentFixture<AvatarComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [ AvatarComponent ]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(AvatarComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});

	it('Should get the initials correctly', () => {
		expect(getInitials('John Doe')).toEqual('JD');
		expect(getInitials('John Doe The Second')).toEqual('JS');
	});
});
