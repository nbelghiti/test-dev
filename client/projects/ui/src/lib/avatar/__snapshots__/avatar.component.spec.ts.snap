// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarComponent Should render correctly 1`] = `
<ui-avatar
  compact="false"
  initials=""
>
  <span
    class="c-avatar"
    ng-reflect-klass="c-avatar"
    ng-reflect-ng-class="[object Object]"
  >
    <abbr
      title="undefined"
    >
      
    </abbr>
  </span>
</ui-avatar>
`;
